pkg/build/webhook/generic/generic.go

```diff
@@ -66,6 +68,9 @@ func (p *WebHookPlugin) Extract(buildCfg *api.BuildConfig, secret, path string,
 			glog.V(4).Infof("Error unmarshaling json %v, but continuing", err)
 			return nil, true, nil
 		}
+		if len(data.Env) > 0 {
+			updateBuildEnv(buildCfg, data.Env)
+		}
```

Comment:
On Tue, Apr 12, 2016 at 3:14 PM, Ben Parees <notifications@github.com>
wrote:

> In pkg/build/webhook/generic/generic.go
> <https://github.com/openshift/origin/pull/8477#discussion_r59436222>:
>
> > @@ -66,6 +68,9 @@ func (p *WebHookPlugin) Extract(buildCfg *api.BuildConfig, secret, path string,
> >  			glog.V(4).Infof("Error unmarshaling json %v, but continuing", err)
> >  			return nil, true, nil
> >  		}
> > +		if len(data.Env) > 0 {
> > +			updateBuildEnv(buildCfg, data.Env)
> > +		}
>
> Well, ultimately the build object is modified, but yes, there is a middle
> man, and it is not modified directly
>
> right, my point was there's no reason to modify the buildconfig when the
> end goal is to produce a build w/ those env vars, not a buildconfig with
> them.
>
> If I read this right, you are saving information in the annotations to use
> later. If so, not 100% sure I like that approach.
>
> i probably should have been more specific about what i wanted you to look
> at :) I was referring to the mechanism i use to add content to the build
> object, not where the content come from. (and also the point in the code at
> which the build object is mutated) My PR is solving a different problem
> than you are (how to let users define annotations and labels as part of a
> build request and have those items get added to the resulting build). That
> is why my PR is using annotations. it's not saving information to
> annotations, the annotations are the information.
>
Gotcha.
 I *believe* the change I'm thinking of will directly update the build
object along the paths existing today and will be along the lines of what
you are talking about.  I hopefully should have something up in the near
term.

> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8477/files/c6a3e2f5ebd4de6e829ec8a7d88ce568b2be7e80#r59436222>
>